project-base/storefront/components/Pages/Search/SearchPageContent.tsx

```diff
@@ -0,0 +1,47 @@
+import { SearchProducts } from './SearchProducts';
+import { LastVisitedProducts } from 'components/Blocks/Product/LastVisitedProducts/LastVisitedProducts';
+import { SkeletonPageSearch } from 'components/Blocks/Skeleton/SkeletonPageSearch';
+import { Webline } from 'components/Layout/Webline/Webline';
+import { SearchContent } from 'components/Pages/Search/SearchContent';
+import { useSearchQuery } from 'graphql/requests/search/queries/SearchQuery.generated';
+import useTranslation from 'next-translate/useTranslation';
+import { usePersistStore } from 'store/usePersistStore';
+import { isClient } from 'utils/isClient';
+import { useCurrentSearchStringQuery } from 'utils/queryParams/useCurrentSearchStringQuery';
+
+export const SearchPageContent: FC = () => {
+    const { t } = useTranslation();
+    const searchString = useCurrentSearchStringQuery();
+    const userIdentifier = usePersistStore((state) => state.userId);
+
+    const [{ data: searchData, fetching }] = useSearchQuery({
+        variables: {
+            search: searchString!,
+            isAutocomplete: false,
+            userIdentifier: userIdentifier!,
+        },
+        pause: !searchString || !isClient || !userIdentifier,
+    });
+
+    if (!searchString) {
+        return (
+            <div className="mb-5 p-12 text-center">
+                <strong>{t('There are no results as you have searched with an empty query...')}</strong>
+            </div>
+        );
+    }
+
+    return (
+        <>
+            <Webline>
+                {(fetching || !isClient) && <SkeletonPageSearch />}
+
+                {!!searchData && !fetching && <SearchContent searchResults={searchData} />}
```

Comment:
We have several places where we do the same conditions. We always will display some content - if we don't have searchData here we don't display the component. But in case of product search there is always at least this content 

![image](https://github.com/shopsys/shopsys/assets/11465991/b3c43485-373c-49a3-a4ea-0e041e20644d)